packages/mui-base/src/TabUnstyled/useTab.ts

```diff
@@ -0,0 +1,87 @@
+import { useTabContext, getTabId, getPanelId } from '../TabsUnstyled';
+import { useButton } from '../ButtonUnstyled';
+
+export interface UseTabProps {
+  /**
+   * You can provide your own value. Otherwise, we fallback to the child position index.
+   */
+  value?: number | string;
+  /**
+   * Callback invoked when new value is being set.
+   */
+  onChange?: (event: React.SyntheticEvent, value: number | string) => void;
+  onClick?: React.MouseEventHandler;
+  disabled?: boolean;
+  onFocus?: React.FocusEventHandler;
+  ref: React.Ref<any>;
+}
+
+const useTab = (props: UseTabProps) => {
+  const { value: valueProp, onChange, onClick, onFocus } = props;
+
+  const { getRootProps: getRootPropsButton, ...otherButtonProps } = useButton(props);
+
+  const context = useTabContext();
+  if (context === null) {
+    throw new TypeError('No TabContext provided');
```

Comment:
I'm not sure if this should be a TypeError as it has nothing to do with types, just specifying the lack of a context.

Reply:
I copied this from the lab's implementation, but I agree. Will change it to Error.